src/rofiles-fuse/main.c

```diff
@@ -529,18 +529,68 @@ callback_access (const char *path, int mode)
   return 0;
 }
 
+static int
+get_fullpath (char *fullpath, size_t fullpathsize, const char *path)
+{
+  char buf[32];
+  ssize_t n;
+
+  if (basefd < 0)
+      return -ENOTSUP;
+
+  snprintf (buf, sizeof(buf), "/proc/self/fd/%d", basefd);
+  n = readlink (buf, fullpath, fullpathsize);
```

Comment:
Any reason to explicitly `readlink` here versus just constructing the path `/proc/self/fd/%d/user/provided/path` and letting the kernel walk it?

Reply:
`/proc/self/fd/3` is an actual link and `readlink` seemed good to get the value of it.

Reply:
Sure, but the kernel will happily traverse it for us just like any other symlink (although it's really a "magic link"); just like you can `ls -al /proc/self/cwd/foo` and `ls` will run directly run e.g.
`statx(AT_FDCWD, "/proc/self/cwd/foo", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_MODE|STATX_NLINK|STATX_UID|STATX_GID|STATX_MTIME|STATX_SIZE, 0x7ffd789d4280) ...`
without walking the symlink explicitly.